engine/src/main/java/org/terasology/rendering/world/WorldRendererLwjgl.java

```diff
@@ -197,101 +266,97 @@ public void render(DefaultRenderingProcess.StereoRenderState stereoRenderState)
         switch (stereoRenderState) {
             case MONO:
                 currentRenderingStage = WorldRenderingStage.DEFAULT;
+                isFirstRenderingStageForCurrentFrame = true;
                 break;
             case OCULUS_LEFT_EYE:
                 currentRenderingStage = WorldRenderingStage.OCULUS_LEFT_EYE;
                 playerCamera.updateFrustum();
+                isFirstRenderingStageForCurrentFrame = true;
                 break;
             case OCULUS_RIGHT_EYE:
                 currentRenderingStage = WorldRenderingStage.OCULUS_RIGHT_EYE;
                 playerCamera.updateFrustum();
+                isFirstRenderingStageForCurrentFrame = false;
                 break;
         }
 
-        if (stereoRenderState == DefaultRenderingProcess.StereoRenderState.MONO || stereoRenderState == DefaultRenderingProcess.StereoRenderState.OCULUS_LEFT_EYE) {
-            renderableWorld.updateAndQueueVisibleChunks();
-        } else {
-            renderableWorld.updateAndQueueVisibleChunks(false, false);
-        }
-
-        if (renderingConfig.isDynamicShadows()
-                && (stereoRenderState == DefaultRenderingProcess.StereoRenderState.MONO || stereoRenderState == DefaultRenderingProcess.StereoRenderState.OCULUS_LEFT_EYE)) {
-            DefaultRenderingProcess.getInstance().beginRenderSceneShadowMap();
-            renderShadowMap();
-            DefaultRenderingProcess.getInstance().endRenderSceneShadowMap();
-        }
+        renderableWorld.updateAndQueueVisibleChunks(isFirstRenderingStageForCurrentFrame);
 
-        DefaultRenderingProcess.getInstance().beginRenderReflectedScene();
-        glCullFace(GL11.GL_FRONT);
-        playerCamera.setReflected(true);
+        renderShadowMap();
         renderWorldReflection();
-        playerCamera.setReflected(false);
-        glCullFace(GL11.GL_BACK);
-        DefaultRenderingProcess.getInstance().endRenderReflectedScene();
 
-        renderWorld();
+        toggleWireFrameON(); // if wireframe is enabled
```

Comment:
I originally thought of calling this method ifEnabledToggleWireFrameOn(). Eventually settled for a shorter method name and the comment. In hindsight I guess I should add the same comment also to renderShadowMap() , renderWorldReflection() and renderChunkRefractiveReflective() or have a similar ifEnabled prefix. Originally the code had those checks in plain sight right in the render() method, but I really would like to keep this method as a clear, simple list of what's happening each frame.